src/hooks/usePopoverRef/components/Container.tsx

```diff
@@ -52,6 +77,23 @@ const PopoverContainer = React.forwardRef<
         }
     );
 
+    const [isInView, setIsInView] = useInViewChecker(ref as React.MutableRefObject<HTMLDivElement>, 4);
+    const [testedPlacements, setTestedPlacements] = React.useState<PopoverPlacement[]>([]);
+    React.useEffect(() => {
+        if (isInView || testedPlacements.length >= 4) {
+            return;
+        }
+
+        const nextPlacement = getAlternativePlacement(
+            preferredPlacement || 'below',
+            testedPlacements
+        );
+
+        setTestedPlacements([...testedPlacements, nextPlacement]);
+        setPlacement(nextPlacement);
+        setIsInView(true);
+    }, [isInView]);

```

Comment:
Missing dependency on `testedPlacements`?

Reply:
No. It should only test a new placement when isInView changes to false, not when testedPlacements changes.